app.rb

```diff
@@ -202,29 +202,34 @@ def write_csv_for_mturk(path, all_assignments, approved_assignments, rejected_as
 assignments_to_approve = Set.new
 assignments_to_reject = {}
 
+columns = {}
+
 is_header = true
 CSV.foreach('batch.csv') do |row|
-  # Skip header row
   if is_header
     is_header = false
+    # Record the index of header columns
+    row.each_with_index do |column_header, index|
+      columns[column_header] = index
+    end
     next
   end
 
-  assignment_id = row[14]
+  assignment_id = row[columns['AssignmentId']]
 
   assignments[assignment_id] = {
-    hit_id: row[0],
-    worker_id: row[15],
-    school: row[27],
-    region: row[28],
-    found_teacher: row[30],
+    hit_id: row[columns['HITId']],
```

Comment:
How would you feel about abstracting these `columns[*]` keys?

The lookup could become: `hit_id: row[COLUMN_HITId]`, with the constant `COLUMN_HITId = columns['HITId']` up the top of this file.